torch/profiler/_pattern_matcher.py

```diff
@@ -69,6 +69,31 @@ def match(self, event):
         return re.search(self.name, event.name()) is not None
 
 
+class ExtraCUDACopyPattern(Pattern):
+
+    def __init__(self, prof: profile):
+        assert prof.with_stack
+        super().__init__(prof)
+        self.description = "Filled a CPU tensor and immediately moved it to GPU. Please initalize it on GPU."
+
+    def match(self, event):
+        if event.name() != "aten::to":
+            return False
+        # Up one level
+        event = event.parent
+        if event is None:
+            return False
+        # Check if we have a aten::fill_ in previous leaf
+        event = self.prev_of(event)
+        if event is None:
+            return False
+        while event.children:
+            event = event.children[-1]
+            if event.name() == "aten::fill_" or event.name() == "aten::zero_":
```

Comment:
I thought `zero_` called into `fill_` to actually implement the compute?

Reply:
It turn out that zero_ is a special case and it dosen't call fill_ :(